### ISSUE TITLE
Hash#each raises ArgumentError with a method of arity 2 converted to a block

### ISSUE THREAD

**Issue by mvz** [jruby/jruby#5896]
### Environment

Provide at least:
* JRuby version (`jruby -v`) and command line (flags, JRUBY_OPTS, etc)
```
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b OpenJDK 64-Bit Server VM 11.0.5-ea+6-post-Debian-2 on 11.0.5-ea+6-post-Debian-2 [linux-x86_64]
JRUBY_OPTS=--dev
JAVA_OPTS=" --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang.reflect=ALL-UNNAMED --add-opens java.base/java.security.cert=ALL-UNNAMED --add-opens java.base/java.security=ALL-UNNAMED --add-opens java.base/javax.crypto=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.nio.channels.spi=ALL-UNNAMED --illegal-access=warn"
```
* Operating system and platform (e.g. `uname -a`)
```
Linux bean 5.2.0-2-amd64 #1 SMP Debian 5.2.9-2 (2019-08-21) x86_64 GNU/Linux
```

### Expected Behavior

Given the following Ruby script:
```ruby
class Bar
  def foo
    { a: 1, b: 2}.each(&method(:bar))
  end

  def bar(a, b)
    puts "#{a}: #{b}"
  end
end

Bar.new.foo
```

CRuby has the following output:
```
a: 1
b: 2
```

### Actual Behavior

JRuby has the following output:
```
ArgumentError: wrong number of arguments (given 1, expected 2)
     bar at foo.rb:3
    each at org/jruby/RubyHash.java:1419
     foo at foo.rb:3
  <main> at foo.rb:11
```

**Comment by eregon** [comment]
Ref: https://github.com/ruby/spec/pull/811#issuecomment-717891062

**Comment by headius** [comment]
As mentioned on #6451, we'll wait a bit and give @marshalium a chance to sort out the right fix and behavior here. The confusing combination is Hash#each plus a proc-ified Map of arity 2; CRuby actually passes two arguments, rather than a splattable Array, but we pass a splattable Array and since our Method is strict (perhaps correctly so) it rejects it as a single argument.

**Comment by headius** [comment]
Fixed by @marshalium in #6451.
